### PR TITLE
L3 / L3 V6  Egress ACL table creation failure

### PR DESCRIPTION
**Pull request by ArthiGovindaraj**
**What I did**
Added checks to ignore the range type qualifier for L3 and L3V6 ACL tables in egress for Broadcom platforms.

**Why I did it**
There are issues seen when creating Egress L3/L3V6 table in Broadcom platforms. Error returned as "Feature not supported". 
Further analysis shows that range qualifiers seem to part of Ingress ACL in Broadcom and the same does not seem to be supported on the egress side. 
So checks are added to avoid the range qualifier in egress side for Broadcom platforms.

**How I verified it**
**Without changes:**
_* Created egress L3/L3 V6 table and ensured that the tables are created without range qualifiers._
 "config acl add table -s egress -p Ethernet0 L3_TABLE L3/L3V6"
Dec  7 06:35:00.262994 sonic ERR syncd#syncd: :- processQuadEvent: attr: SAI_ACL_TABLE_ATTR_FIELD_SRC_IP: true
Dec  7 06:35:00.262994 sonic ERR syncd#syncd: :- processQuadEvent: attr: SAI_ACL_TABLE_ATTR_FIELD_DST_IP: true
Dec  7 06:35:00.262994 sonic ERR syncd#syncd: :- processQuadEvent: attr: SAI_ACL_TABLE_ATTR_FIELD_OUTER_VLAN_ID: true
Dec  7 06:35:00.262994 sonic ERR syncd#syncd: :- processQuadEvent: attr: SAI_ACL_TABLE_ATTR_FIELD_L4_SRC_PORT: true
Dec  7 06:35:00.262994 sonic ERR syncd#syncd: :- processQuadEvent: attr: SAI_ACL_TABLE_ATTR_FIELD_L4_DST_PORT: true
Dec  7 06:35:00.263028 sonic ERR syncd#syncd: :- processQuadEvent: attr: SAI_ACL_TABLE_ATTR_FIELD_ETHER_TYPE: true
Dec  7 06:35:00.263028 sonic ERR syncd#syncd: :- processQuadEvent: attr: SAI_ACL_TABLE_ATTR_FIELD_IP_PROTOCOL: true
Dec  7 06:35:00.263055 sonic ERR syncd#syncd: :- processQuadEvent: attr: SAI_ACL_TABLE_ATTR_FIELD_TCP_FLAGS: true
Dec  7 06:35:00.263055 sonic ERR syncd#syncd: :- processQuadEvent: attr: SAI_ACL_TABLE_ATTR_FIELD_ACL_IP_TYPE: true
Dec  7 06:35:00.263108 sonic ERR syncd#syncd: :- processQuadEvent: attr: SAI_ACL_TABLE_ATTR_FIELD_ICMP_TYPE: true
Dec  7 06:35:00.263108 sonic ERR syncd#syncd: :- processQuadEvent: attr: SAI_ACL_TABLE_ATTR_FIELD_ICMP_CODE: true
Dec  7 06:35:00.263108 sonic ERR syncd#syncd: :- processQuadEvent: attr: SAI_ACL_TABLE_ATTR_FIELD_ACL_RANGE_TYPE: 2:SAI_ACL_RANGE_TYPE_L4_SRC_PORT_RANGE,SAI_ACL_RANGE_TYPE_L4_DST_PORT_RANGE
Dec  7 06:35:00.263108 sonic ERR syncd#syncd: :- processQuadEvent: attr: SAI_ACL_TABLE_ATTR_ACL_ACTION_TYPE_LIST: 2:SAI_ACL_ACTION_TYPE_PACKET_ACTION,SAI_ACL_ACTION_TYPE_COUNTER
Dec  7 06:35:00.263131 sonic ERR syncd#syncd: :- processQuadEvent: attr: SAI_ACL_TABLE_ATTR_ACL_STAGE: SAI_ACL_STAGE_EGRESS
Dec  7 06:35:00.263329 sonic ERR swss#orchagent: :- create: create status: SAI_STATUS_NOT_SUPPORTED
Dec  7 06:35:00.263398 sonic ERR swss#orchagent: :- addAclTable: Failed to create ACL table L3EGR

**With the changes:**
Egress table created with qualifiers srcip, dst ip, outer vlan id, l4 src port, l4 dst port, ether type, ip protocol, tcp flags, ip type, outport and icmp typecode.
show acl table
Name    Type    Binding     Description    Stage
L3EGR   L3      Ethernet46  L3EGR          egress

* Created ingress L3/L3 V6 tables and ensured that tables are created with range qualifiers.
"config acl add table -s ingress -p Ethernet0 L3_TABLE L3/L3V6"
root@sonic:~# show acl table
Name    Type    Binding     Description    Stage
L3EGR   L3      Ethernet46  L3EGR          egress
L3_ACL  L3      Ethernet0   L3_ACL         ingress => includes range type qualifier.

The changes were validated in TD3 platform. 

**Details if related**

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [ ] 202106
- [ ] 202111
- [x] 202205

### CODE REVIEW

File: orchagent/aclorch.cpp [hunk 1]
```diff
@@ -2097,10 +2153,17 @@ bool AclTable::addStageMandatoryMatchFields()
...
+                if (match != SAI_ACL_TABLE_ATTR_FIELD_ACL_RANGE_TYPE)
+                {
+                    type.addMatch(make_shared<AclTableMatch>(match));
+                    SWSS_LOG_INFO("Added mandatory match field %s for table type %s stage %d",
+                        sai_serialize_enum(match, &sai_metadata_enum_sai_acl_table_attr_t).c_str(),
+                        type.getName().c_str(), stage);
+                }
+                else
+                {
+                    addStageMandatoryRangeFields();
```
Comment (bingwang-ms) [inline orchagent/aclorch.cpp:2165]: Is it possible that `addStageMandatoryRangeFields` is called for multiple times when walking through the for loop?

**Comment by bingwang-ms** [comment]
The code change looks good to me. Please add vstest cases to cover the change. Thanks

**Comment by ArthiGovindaraj** [comment]
> The code change looks good to me. Please add vstest cases to cover the change. Thanks
Thanks for the review. I will update the coverage test cases.

**Comment by ArthiGovindaraj** [comment]
@bingwang-ms The VS test cases are added and coverage has passed.

**Review by bingwang-ms — APPROVED** [review]

**Comment by ArthiGovindaraj** [comment]
Hi Sunny, 
Can you approve/merge these changes.

**Pull request merged**